include/exadg/incompressible_navier_stokes/spatial_discretization/operator_consistent_splitting.cpp

```diff
@@ -0,0 +1,776 @@
+    {      
+      dealii::Point<dim, scalar> q_points = integrator.quadrature_point(q);
+
+      // evaluate right-hand side
+      vector rhs =
+        FunctionEvaluator<1, dim, Number>::value(*(this->field_functions->right_hand_side),
+                                                  q_points,
+                                                  this->evaluation_time);
+
+      integrator.submit_gradient(rhs, q);
+    }
+    integrator.integrate_scatter(dealii::EvaluationFlags::gradients, dst);
+  }
+}
+
+
+
+
+template<int dim, typename Number>
+void
+OperatorConsistentSplitting<dim, Number>::local_rhs_ppe_div_term_body_forces_inner_face(
+  dealii::MatrixFree<dim, Number> const & matrix_free,
+  VectorType &                            dst,
+  VectorType const &,
+  Range const & face_range) const
+{
+  unsigned int dof_index_pressure  = this->get_dof_index_pressure();
+  unsigned int quad_index_pressure = this->get_quad_index_pressure();
+
+  FaceIntegratorP pressure_minus(matrix_free, true, dof_index_pressure, quad_index_pressure);
+  FaceIntegratorP pressure_plus(matrix_free, false, dof_index_pressure, quad_index_pressure);
+
+  for(unsigned int face = face_range.first; face < face_range.second; face++)
+  {
+    pressure_minus.reinit(face);
+    pressure_plus.reinit(face);
+
+    for(unsigned int q = 0; q < pressure_minus.n_q_points; ++q)
+    {      
+      dealii::Point<dim, scalar> q_points = pressure_minus.quadrature_point(q);
+
+      // evaluate right-hand side
+      vector rhs =
+        FunctionEvaluator<1, dim, Number>::value(*(this->field_functions->right_hand_side),
+                                                  q_points,
+                                                  this->evaluation_time);
+      const auto normal = pressure_minus.normal_vector(q);
+      const auto flux   = rhs * normal;
+
+      pressure_minus.submit_value(-flux, q);
+      pressure_plus.submit_value(flux, q);
+    }
+    pressure_minus.integrate_scatter(dealii::EvaluationFlags::values, dst);
+    pressure_plus.integrate_scatter(dealii::EvaluationFlags::values, dst);
+  }
+}
+
+
+template<int dim, typename Number>
+void
+OperatorConsistentSplitting<dim, Number>::local_rhs_ppe_div_term_body_forces_boundary_face(
+  dealii::MatrixFree<dim, Number> const & matrix_free,
+  VectorType &                            dst,
+  VectorType const &,
+  Range const & face_range) const
+{
+  unsigned int dof_index_pressure  = this->get_dof_index_pressure();
+  unsigned int quad_index_pressure = this->get_quad_index_pressure();
+
+  FaceIntegratorP integrator(matrix_free, true, dof_index_pressure, quad_index_pressure);
+
+  for(unsigned int face = face_range.first; face < face_range.second; face++)
+  {
+    integrator.reinit(face);
+
+    BoundaryTypeU boundary_type =
+      this->boundary_descriptor->velocity->get_boundary_type(matrix_free.get_boundary_id(face));
+
+    for(unsigned int q = 0; q < integrator.n_q_points; ++q)
```

Comment:
Since we do nothing except for the Neumann case, can you reorder the `if` and the `for` loop?
Then we can skip the `integrator.submit_value(zero, q);` for the Dirichlet faces.
Also this is similar to https://github.com/exadg/exadg/pull/792#discussion_r2478369193